packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -651,31 +649,31 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
   }
 
   @override
-  bool hitTestChildren(BoxHitTestResult result, { Offset position }) {
+  bool hitTestChildren(BoxHitTestResult result, { Offset? position }) {
     double mainAxisPosition, crossAxisPosition;
     switch (axis) {
       case Axis.vertical:
-        mainAxisPosition = position.dy;
+        mainAxisPosition = position!.dy;
         crossAxisPosition = position.dx;
         break;
       case Axis.horizontal:
-        mainAxisPosition = position.dx;
+        mainAxisPosition = position!.dx;
         crossAxisPosition = position.dy;
         break;
     }
     assert(mainAxisPosition != null);
     assert(crossAxisPosition != null);
     final SliverHitTestResult sliverResult = SliverHitTestResult.wrap(result);
     for (final RenderSliver child in childrenInHitTestOrder) {
-      if (!child.geometry.visible) {
+      if (!child.geometry!.visible) {
         continue;
       }
       final Matrix4 transform = Matrix4.identity();
       applyPaintTransform(child, transform);
       final bool isHit = result.addWithPaintTransform(
         transform: transform,
         position: null, // Manually adapting from box to sliver position below.
-        hitTest: (BoxHitTestResult result, Offset _) {
+        hitTest: (BoxHitTestResult result, Offset? _) {
```

Comment:
applyPaintTransform(child, transform); // must be invertible
      final bool isHit = result.addWithOutOfBandPosition(
        paintTransform: transform,
        hitTest: (BoxHitTestResult result) {